design/src/rfcs/rfc0037_retry_classifier_customization.md

```diff
@@ -0,0 +1,240 @@
+# RFC: User-configurable retry classification
+
+> Status: RFC
+>
+> Applies to: client
+
+For a summarized list of proposed changes, see the [Changes Checklist](#changes-checklist) section.
+
+This RFC defines the user experience and implementation of user-configurable retry classification. Custom retry classifiers enable users to change what errors are retried while still allowing them to rely on defaults set by SDK authors when desired.
+## Terminology
+
+- **Smithy Service**: An HTTP service, whose API is modeled with the [Smithy IDL](https://www.smithy.io).
+- **Smithy Client**: An HTTP client generated by smithy-rs from a `.smithy` model file.
+- **AWS SDK**: A **smithy client** that's specifically configured to work with an AWS service.
+- **Operation**: A modeled interaction with a service, defining the proper input and expected output types, as well as important metadata related to request construction. "Sending" an operation implies sending one or more HTTP requests to **smithy service**, and then receiving an output or error in response.
+- **Orchestrator**: The client code which manages the request/response pipeline. The orchestrator is responsible for: 
+	- Constructing, serializing, and sending requests.
+	- Receiving, deserializing, and (optionally) retrying requests.
+	- Running interceptors *(not covered in this RFC)* and handling errors.
+- **Runtime Component**: A part of the orchestrator responsible for a specific function. Runtime components are *always* required and may depend on specific configuration. Examples include the endpoint resolver, retry strategy, and request signer.
+- **Runtime Plugin**: Code responsible for setting and **runtime components** and related configuration. Runtime plugins defined by codegen are responsible for setting default configuration and altering the behavior of **smithy clients** including the **AWS SDKs**.
+- **Retry Strategy**: The process by which the orchestrator determines when and how to retry failed requests. Only one retry strategy may be set. The retry strategy depends upon the **retry classifier** to interpret responses and determine if they are retryable.
+- **Retry Classifier**: Code responsible for introspecting the request/response pipeline's and determining if a retry is necessary. Multiple retry classifiers may be combined into a **retry classifier chain**.
+- **Retry Classifier Chain**: Requests can fail for any number of reasons, but retry classifiers are specific, meaning that they target a narrow range of possibilities. Rather than define a single retry classifier to handle all possible outcomes, multiple classifiers are chained together. Each classifier is run in turn, and the chain ends when a classifier decides a response is retryable, or a classifier may determine that a response must not be retried, or after all classifiers have run and haven't decided a retry is necessary.
```

Comment:
I think the retry classifier pieces should be defined as part of the RFC rather than the terminology for making sense of the RFC.